fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ElectionHandlerTest.java

```diff
@@ -199,4 +202,25 @@ public void testHandleElectionEnd() throws DataHandlingException {
     assertTrue(electionOpt.isPresent());
     assertEquals(EventState.CLOSED, electionOpt.get().getState());
   }
+
+  @Test
+  public void testHandleOpenElection() throws DataHandlingException {
+    OpenElection openElection = new OpenElection(lao.getId(), election.getId(), openedAt);
+    MessageGeneral message = new MessageGeneral(SENDER_KEY, openElection, GSON);
+
+    for (EventState state : EventState.values()) {
+      election.setEventState(state);
+
+      if (state == EventState.CREATED) {
+
+        //If the previous state was CREATED, it should change it to OPENED.
+        messageHandler.handleMessage(laoRepository, messageSender,
+            election.getChannel(), message);
```

Comment:
Don't you want to send the message for each state and not only for created state ?

Reply:
It should transition from CREATED --> OPENED only when the current state is CREATED, this way I need to check for other states as well

Reply:
I think you should move the line `messageHandler.handleMessage(laoRepository, messageSender, election.getChannel(), message);` should be moved outside the loop.

So the test makes sure that the election isn't opened when the state is not create.